src/gerbil/runtime/mop.ss

```diff
@@ -802,29 +874,26 @@ namespace: #f
         (type-descriptor-methods-set! klass vtab)
         (type-descriptor-seal! klass)))))
 
+;; NB: 1. This implementation has quadratic complexity, and
+;; 2. it relies on methods using next-method to be actually location dependent
+;; and NOT things you can share between different classes.
+;; Changing the protocol to access an explicit super argument would be semantically nicer
+;; and would enable linear complexity for next-method (in the number of classes,
+;; or even just in the number of applicable method, if resolved only once),
+;; but would be somewhat incompatible.
```

Comment:
this is not the whole story, so this is rather misleading. It is a tradeoff between have fast next-method (which is very rarely used) vs efficiency everywhere else.

Reply:
Better?